python/cudf/cudf/tests/test_multiindex.py

```diff
@@ -781,13 +781,16 @@ def test_multiindex_copy_sem(data, levels, codes, names):
         ),
     ],
 )
+@pytest.mark.parametrize("copy_on_write", [True, False])
 @pytest.mark.parametrize("deep", [True, False])
-def test_multiindex_copy_deep(data, deep):
+def test_multiindex_copy_deep(data, copy_on_write, deep):
     """Test memory identity for deep copy
     Case1: Constructed from GroupBy, StringColumns
     Case2: Constructed from MultiIndex, NumericColumns
     """
-    same_ref = not deep
+    original_cow_setting = cudf.get_option("copy_on_write")
+    cudf.set_option("copy_on_write", copy_on_write)
+    same_ref = (not deep) or (cudf.get_option("copy_on_write") and not deep)
```

Comment:
This var seems to no longer be used in the assertion below. Is that intentional/correct? If so, can this line be deleted?

Reply:
It is being used in the `elif` block below. Moved it into the block.